Microsoft.Toolkit.Uwp.UI.Controls/InAppNotification/InAppNotification.cs

```diff
@@ -251,22 +258,36 @@ private void UpdateContent(NotificationOptions notificationOptions)
         /// <param name="notificationOptions">Information about the notification to display</param>
         private void Show(NotificationOptions notificationOptions)
         {
-            bool shouldDisplayImmediately = ShouldDisplayImmediately();
-
-            if (StackMode == StackMode.QueueBehind)
+            var shouldDisplayImmediately = true;
+            switch (StackMode)
             {
-                _stackedNotificationOptions.Add(notificationOptions);
-            }
-
-            if (StackMode == StackMode.StackInFront)
-            {
-                _stackedNotificationOptions.Insert(0, notificationOptions);
+                case StackMode.Replace:
+                    _stackedNotificationOptions.Clear();
+                    _stackedNotificationOptions.Add(notificationOptions);
+                    break;
+                case StackMode.StackInFront:
+                    _stackedNotificationOptions.Insert(0, notificationOptions);
+                    break;
+                case StackMode.QueueBehind:
+                    _stackedNotificationOptions.Add(notificationOptions);
+                    shouldDisplayImmediately = _stackedNotificationOptions.Count == 1;
+                    break;
+                default:
+                    break;
             }
 
             if (shouldDisplayImmediately)
             {
                 UpdateContent(notificationOptions);
-                Show(notificationOptions.Duration);
+
+                Visibility = Visibility.Visible;
+                VisualStateManager.GoToState(this, StateContentVisible, true);
+
+                if (notificationOptions.Duration > 0)
```

Comment:
I noticed that we never reset the time, so if you're in stacked/queue mode and submit queries in rapid succession they don't get equal time or you might miss one. should we restructure this a bit here or do you want me to open a new issue?

You can repro this behavior in the sample app by setting the Notification Duration to a value like 500-1000 and then clicking the 'Show notification with random text' button 3 times, but you'll only see 2 messages.

Reply:
I've noticed this too when using the "show with text" button. Since we don't have a lot of strings, we are often seeing the same one and it can be misleading.
Using the "show with object" button is better in this case since we're are more likely to get a different content change in the notifications.

I've added some traces to double check and everything seems to behave as expected. This only missing part I've found is a missing `_dismissTimer.Stop()` if  `notificationOptions.Duration <= 0` when immediately pushing a new notification.

Reply:
There's still a bug here, but I'll open another issue as it's unrelated to this change and I think a bit more complex than I think (see https://github.com/windows-toolkit/WindowsCommunityToolkit/issues/3390). The issue isn't necessarily with stopping the timer, it's that we don't extend the timer when we flip to the next message in the queue/stack.